lib/Doctrine/ORM/Query/Expr/Join.php

```diff
@@ -127,6 +127,6 @@ public function __toString()
         return strtoupper($this->joinType) . ' JOIN ' . $this->join
              . ($this->alias ? ' ' . $this->alias : '')
              . ($this->indexBy ? ' INDEX BY ' . $this->indexBy : '')
-             . ($this->condition ? ' ' . strtoupper($this->conditionType) . ' ' . $this->condition : '');
+             . ($this->condition && $this->conditionType !== null ? ' ' . strtoupper($this->conditionType) . ' ' . $this->condition : '');
```

Comment:
I think `$this->conditionType` is supposed to be not null at this point, but I still want to mention that this is a change of behaviour. I still approve it in this PR because it's a smaller break.

Or maybe this line should be merged into 2.10 instead of 2.9?

Reply:
I agree, this line should be reverted.

Reply:
I added this check because this Psalm finding:
```
ERROR: PossiblyNullArgument - lib/Doctrine/ORM/Query/Expr/Join.php:130:53 - Argument 1 of strtoupper cannot be null, possibly null value provided (see https://psalm.dev/078)
             . ($this->condition ? ' ' . strtoupper($this->conditionType) . ' ' . $this->condition : '');
```
Now I reverted the check based on your suggestion and used a inline docblock instead.

Reply:
The CS rules are not allowing an inline docblock. They suggest to use assertion, but the assertion causes the method to throw an exception if it is not fulfilled.
What do you propose for this case?

Reply:
Keep the error baselined and resolve it on 2.10 with an `assert()`.

Reply:
Done.